src/odemis/driver/npmc.py

```diff
@@ -0,0 +1,809 @@
+        future (Future): the future it handles
+        pos (dict str -> float): axis name -> relative target position
+        raise:
+            ValueError: if the target position is
+            TMCLError: if the controller reported an error
+            CancelledError: if cancelled before the end of the move
+        """
+        with future._moving_lock:
+
+            end = 0  # expected end
+            moving_axes = set()
+            for an, v in pos.items():
+                aid = self._axis_map[an]
+                moving_axes.add(aid)
+                self.MoveRelPos(aid, v * self._axis_conv_factor[aid])
+                # compute expected end
+                # convert to mm units
+                dur = driver.estimateMoveDuration(abs(v) * self._axis_conv_factor[aid],
+                                self._speed[an],
+                                self._accel[an])
+                    
+                end = max(time.time() + dur, end)
+
+            self._waitEndMove(future, moving_axes, end)
+        self.checkError()
+        logging.debug("move successfully completed")
+
+    def _doMoveAbs(self, future, pos):
+        """
+        Blocking and cancellable absolute move
+        future (Future): the future it handles
+        pos (dict str -> float): axis name -> absolute target position
+        raise:
+            TMCLError: if the controller reported an error
+            CancelledError: if cancelled before the end of the move
+        """
+        with future._moving_lock:
+            end = 0  # expected end
+            old_pos = self._applyInversion(self.position.value)
+            moving_axes = set()
+            for an, v in pos.items():
+                aid = self._axis_map[an]
+                moving_axes.add(aid)
+                self.MoveAbsPos(aid, v * self._axis_conv_factor[aid])
+                d = abs(v - old_pos[an])
+                # convert displacement unit to mm
+                dur = driver.estimateMoveDuration(d * self._axis_conv_factor[aid],
+                                                  self._speed[an],
+                                                  self._accel[an])
+                end = max(time.time() + dur, end)
+
+            self._waitEndMove(future, moving_axes, end)
+        self.checkError()
+        logging.debug("move successfully completed")
+
+    def _waitEndMove(self, future, axes, end=0):
+        """
+        Wait until all the given axes are finished moving, or a request to
+        stop has been received.
+        future (Future): the future it handles
+        axes (set of int): the axes IDs to check
+        end (float): expected end time
+        raise:
+            TimeoutError: if took too long to finish the move
+            CancelledError: if cancelled before the end of the move
+        """
+        moving_axes = set(axes)
+
+        last_upd = time.time()
+        dur = max(0.01, min(end - last_upd, 60))
+        max_dur = dur * 5 + 1
```

Comment:
Why is `max_dur` so big? Either revert back to `* 2` now that the move duration is properly estimated, or add a comment explaining why the estimation is so bad.